src/plugins/gmx/gmxplugin.cpp

```diff
@@ -177,8 +177,10 @@ bool GmxPlugin::write(const Map *map, const QString &fileName)
                     stream.writeAttribute("w", QString::number(tile->width()));
                     stream.writeAttribute("h", QString::number(tile->height()));
 
-                    stream.writeAttribute("xo", QString::number(tile->id() % tile->tileset()->columnCount() * tile->tileset()->tileWidth()));
-                    stream.writeAttribute("yo", QString::number((int)(tile->id() / tile->tileset()->columnCount()) * tile->tileset()->tileWidth()));
+                    int xInTilesetGrid = tile->id() % tile->tileset()->columnCount();
+                    int yInTilesetGrid = (int)(tile->id() / tile->tileset()->columnCount());
+                    stream.writeAttribute("xo", QString::number(tile->tileset()->margin() + (xInTilesetGrid * tile->tileset()->tileSpacing()) + (xInTilesetGrid * tile->tileset()->tileWidth())));
+                    stream.writeAttribute("yo", QString::number(tile->tileset()->margin() + (yInTilesetGrid * tile->tileset()->tileSpacing()) + (yInTilesetGrid * tile->tileset()->tileHeight())));
```

Comment:
Please simplify it a little by doing `tile->tileset()->tileHeight() + tile->tileset()->tileSpacing()` and multiplying by `yInTilesetGrid` once.

Multiplying by tile width for y must just have been a mistake. Please edit your commit message to avoid asking about it and just pointing out you fixed that as well. :-)